src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Tree/Dependencies/Subscriptions/RuleHandlers/DependenciesRuleHandlerBase.cs

```diff
@@ -181,6 +166,10 @@ void HandleChangesForRule(bool resolved, IProjectChangeDescription evaluationPro
 
             bool isImplicit = IsImplicit(projectFullPath, evaluationProperties);
 
+            // When we only have evaluation item, mark the dependency as resolved if we currently have a resolved item improve
+            // this, so that if a resolved item is known, we don't revert back to unresolved state.
+            isResolved = isResolved || changesBuilder.HasResolvedItem(targetFramework, ProviderType, originalItemSpec);
```

Comment:
nit: Is short-circuiting required here? If not, you can do bitwise OR instead.
```suggestion
            isResolved |= changesBuilder.HasResolvedItem(targetFramework, ProviderType, originalItemSpec);
```

Reply:
Not "required", but I prefer to avoid unneeded work where possible.